include/ppx/geometry.h

```diff
@@ -146,8 +153,11 @@ class Geometry
     {
     public:
         Buffer() {}
-        Buffer(BufferType type, uint32_t elementSize)
-            : mType(type), mElementSize(elementSize) {}
+        Buffer(BufferType type, uint32_t elementSize, uint32_t maxElementCount = 0)
+            : mType(type), mElementSize(elementSize), mMaxElementCount(maxElementCount)
```

Comment:
I think I had the name as "mKnownElementCount / mFinalElementCount" at some point earlier to try and mitigate that confusion, but changed it to this for consistency. 

I don't actually want to prevent people from calling Append if the value is not zero, I think there could be cases where the final size is known but the user still wants to use Append functions.

Reply:
`mMaxElementCount` here means 2 things:
- the reserved size in bytes if it is not 0
- a bool indicate the reserved size is not used if it is 0
But if the user is not familiar with this class, by purely looking at header, the 2nd one above is not obvious, and imagine in the future if someone wants to get the max elem count, adding a new public functions to get the mMaxElementCount would be wrong. 

I guess if you don't keep `mMaxElementCount` as a member, there won't be a consistency issue? 

Let's say if `mMaxElementCount` is not zero, If people call `Append()' after `mData` reaching the max size, and there will be no resize (because `mMaxElementCount != 0`), and the source value would be copied to somewhere out of the range which might cause a mem corruption?